### PR TITLE
[Feature] Add notes field to assessment results

### LINKED ISSUE #8729: ✨ Add field `notesForThisAssessment` to `AssessmentResult`
## ✨ Feature

A field is needed to properly implement the assessment modal set. 

## 🕵️ Details

Field `notesForThisAssessment`, schema type string database type text? 

Backend -> Add to database, schema, model, mutations. Possibly validation too as it is a required field 
Frontend -> Update modal to replace placeholder seen below

image

## ✅ Acceptance Criteria

- [ ] Added to wherever specified above

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #8729 

## 👋 Introduction

Adds the notes test column to the assessment results table and dialog form.

## 🧪 Testing

1. Refresh API `docker-compose run --rm maintenance bash refresh_api.sh`
2. Build `npm run dev`
3. Run storybook `npm run storybook:web`
4. Navigagte to the "AssessmentDescisionDialog -> Generic" story
5. Fill out some notes
6. Submit
7. Confirm they appear in the action data
8. Seed the database `docker-compose exec -w /home/site/wwwroot/api webserver sh -c "php artisan migrate:fresh --seed"`
9. Open the database and find an assessment result with notes (or insert one) and copy the pool candidate ID
10. Navigate to `/graphiql`
11. Run a query for the notes using the pool candidate ID copied in step 9
12. Confirm it shows up in the result

```graphql
query AssessmentResults{
  poolCandidate(id: "poolCandidateId") {
    id
    assessmentResults {
      assessmentNotes
    }
  }
}
```

### CODE REVIEW

File: api/database/migrations/2024_01_04_144452_add_notes_column_assessment_results_table.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('assessment_results', function (Blueprint $table) {
+            $table->text('notes')->nullable()->default(null);
+        });
```
Comment (vd1992) [inline api/database/migrations/2024_01_04_144452_add_notes_column_assessment_results_table.php:16]: I'm not 100% on shortening the field name to just "notes", I like a more descriptive name. Given this model already has two notes fields, `other_justification_notes` and `skill_decision_notes` I think a notes field just called "notes" isn't the clearest. 

"notesForThisAssessment", "notesForAssessment", "assessmentNotes" as examples of names. I think they make it clear which input the field lines up with.

**Review by vd1992 — COMMENTED** [review]
Think it works well but I got a thought on naming below

File: api/database/migrations/2024_01_04_144452_add_notes_column_assessment_results_table.php (hunk 1 above)
Comment (esizer) [reply api/database/migrations/2024_01_04_144452_add_notes_column_assessment_results_table.php:16]: Okay, that makes sense. I didn't realize we had so many variations of notes for a single model 😛

**Review by vd1992 — APPROVED** [review]
Great!

**Pull request merged**